Formula/postgresql.rb

```diff
@@ -100,18 +100,20 @@ def post_install
   end
 
   def caveats; <<-EOS.undent
-    If builds of PostgreSQL 9 are failing and you have version 8.x installed,
+    If builds of PostgreSQL 10 are failing and you have version 8.x installed,
     you may need to remove the previous version first. See:
       https://github.com/Homebrew/legacy-homebrew/issues/2510
 
     To migrate existing data from a previous major version (pre-9.0) of PostgreSQL, see:
```

Comment:
Most likely, needs change to `pre-10.0`

Reply:
This comment was already wrong before, because both eg. 8.4 and 9.0 are major versions.

The link points to documentation for upgrading from a any previous version of PostgreSQL, including `pg_upgrade`, which ist supported since PostgreSQL 8.4 and `pg_dumpall` which works with current and older versions of PostgreSQL, but is a more manual process and cannot do in-place upgrades (it's a manual backup and restore).

Quoting https://www.postgresql.org/docs/10/static/pgupgrade.html:

> pg_upgrade supports upgrades from 8.4.X and later to the current major release of PostgreSQL, including snapshot and beta releases.

The conclusion to this is, that for upgrades from anything prior to 8.4, one would have to di a manual `pg_dumpall` on the old cluster and import that dump on the new cluster.

Because the top-level page linked here, contains links to both `pg_upgrade` and `pg_dumpall` methods for upgrading as well as descriptions of when to use which method, we could reduce the comments to link to this single upgrade page.

Reply:
:+1: